tests/testthat/test-data-combined.R

```diff
@@ -0,0 +1,1072 @@
+dataSet <- loadDataSetsFromExcel(
+  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012.xlsx"),
+  importerConfiguration = DataImporterConfiguration$new(getTestDataFilePath("ImporterConfiguration.xml"))
+)
+
+# same as dataSet, but with deliberately changed entries for testing
+dataSet2 <- loadDataSetsFromExcel(
+  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012v2.xlsx"),
+  importerConfiguration = DataImporterConfiguration$new(getTestDataFilePath("ImporterConfiguration.xml"))
+)
+
+# empty initialization ----------------------------
+
+test_that("dataCombined - initialization", {
+  # check class generator
+  expect_true(R6::is.R6Class(DataCombined))
+
+  # initialize empty object
+  myCombDat <- DataCombined$new()
+
+  # the active bindings should all be NULL at this stage
+  expect_null(myCombDat$groupMap)
+  expect_null(myCombDat$names)
+  expect_null(myCombDat$toDataFrame())
+  expect_output(print(myCombDat), "DataCombined:")
+
+  # can't use active bindings like this
+  expect_error(myCombDat$groupMap("x"))
+  expect_error(myCombDat$names("x"))
+  expect_error(myCombDat$dataTransformations("x"))
+
+  # can't enter a list of `SimulationResults` objects
+  expect_error(myCombDat$addSimulationResults(list("x", "y")))
+
+  # can enter a list, but only of `DataSet` objects
+  expect_error(myCombDat$addDataSets(list("x", "y")))
+
+  # `NULL`s not allowed
+  expect_error(myCombDat$addDataSets(list(NULL)))
+  expect_error(myCombDat$addSimulationResults(list(NULL)))
+})
+
+# either `DataSet` and `SimulationResults` provided -------------
+
+test_that("dataCombined - either dataSet or SimulationResults provided", {
+  # only DataSet input
+  myCombDat <- DataCombined$new()
+  myCombDat$addDataSets(dataSet[[1]])
+
+  # data transformations should already be setup with the defaults
+  expect_equal(
+    myCombDat$dataTransformations,
+    structure(
+      list(
+        name = "Stevens_2012_placebo.Placebo_total",
+        xOffsets = 0,
+        yOffsets = 0,
+        xScaleFactors = 1,
+        yScaleFactors = 1
+      ),
+      class = c("tbl_df", "tbl", "data.frame"),
+      row.names = c(NA, -1L)
+    )
+  )
+
+
+  # checking dataframe methods
+  df <- myCombDat$toDataFrame()
+  expect_s3_class(df, "data.frame")
+  expect_equal(dim(df), c(12L, 20L))
+
+  expect_equal(
+    names(df),
+    c(
+      "name", "group", "dataType", "xValues", "xUnit", "xDimension",
+      "yValues", "yUnit", "yDimension", "yErrorValues", "yErrorType",
+      "yErrorUnit", "molWeight", "lloq", "Source", "Sheet", "Organ",
+      "Compartment", "Molecule", "Group Id"
+    )
+  )
+  expect_equal(rep(NA_character_, length(df$group)), df$group)
+  expect_equal(unique(df$name), names(dataSet)[[1]])
+
+  # only SimulationResults input
+
+  # selecting all paths
+  myCombDat2 <- DataCombined$new()
+  myCombDat2$addSimulationResults(simResults)
+
+  # data transformations should already be setup with the defaults
+  expect_equal(
```

Comment:
I am not sure if this is how tests are written in R so I am just commenting from otjher languages point of view.
WE typically try to not test too much at the same time. So one test_that should have only one (a few) expect

If you are changing the context (new dataSets, adding removing etc.. ) you are testing something else. 

You can also use the keyword context() to group tests that belong together. Then in your test that, you can be more explicit about what you are testing. 

Example: Instead of a comment "data transformations should already be setup with the defaults"

This shoulld be the name of the test

test_that("data transformations should already be setup with the defaults")

does it make sense?